### PR TITLE
Single-sided buffer

### PR DESCRIPTION
**Pull request by stev-0**
See discussion at https://lists.osgeo.org/pipermail/postgis-devel/2018-January/026852.html

This is built on top of the trunk code - there is another pull request for the 2.4 branch

### CODE REVIEW

File: regress/Makefile.in [hunk 1]
```diff
@@ -80,6 +80,7 @@ TESTS = \
 	bestsrid \
 	binary \
 	boundary \
+	buffer \
```
Comment (dbaston) [inline regress/Makefile.in]: Can the new tests be put into the existing `regress_buffer_params` file?

**Comment by dbaston** [comment]
Can you please add an entry to the `NEWS` file for this new feature? I think this will only get merged into trunk, not 2.4 (since it's not a bug fix).

**Comment by Komzpa** [comment]
@stev-0 
travis barks on trailing spaces, 
```
./postgis/lwgeom_geos.c:958:				if ( !strcmp(val, "both") ) 
./postgis/lwgeom_geos.c:962:				else if ( !strcmp(val, "left") ) 
```

another failure is caused by (now fixed) median robustness test, so just `git pull; git merge upstream/svn-trunk` will fix it. I'd do it myself it you ticked "allow edits from maintainers", without it I have to ask you to perform it :)

Sorry for long wait, were off to pgconf.ru :)

**Comment by dbaston** [comment]
This works well and I'm planning to merge it in shortly. I'm going to change it only to add error-checking to the new GEOS calls (currently the return values are ignored)

**Comment by Komzpa** [comment]
This PR got merged to svn by @dbaston at https://github.com/postgis/postgis/commit/d884ff8fc9876398f0b3132562f8a4f1f0f71f54

@stev-0 thanks for the contribution! :)

@dbaston there's a bot that will close PR if you write "Closes: https://github.com/postgis/postgis/pull/204" in commit message, otherwise it's left hanging :)

I'll take another look how codecov went around the block and posted the comment.

**Pull request closed without merging**